modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java

```diff
@@ -97,7 +97,7 @@ public GoClientCodegen() {
         typeMapping.put("boolean", "bool");
         typeMapping.put("string", "string");
         typeMapping.put("UUID", "string");
-        typeMapping.put("date", "time.Time");
+        typeMapping.put("date", "SwaggerDateType");
```

Comment:
That is a possibility. 

I have an app in production using the date type so it should help determine any issues that arise from using string.

I will test that tonight and convert if needed.